src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMA = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFromFile(fromFileName);
+        String report = createReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    public String readFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName));) {
+            String line = bufferedReader.readLine();
+            do {
+                stringBuilder.append(line).append(COMA);
+                line = bufferedReader.readLine();
+            } while (line != null);
+        } catch (IOException e) {
+            throw new RuntimeException("Cant read file " + fromFileName, e);
+        }
+        return stringBuilder.toString();
+    }
+
+    public String createReport(String stringFromFile) {
+        int supplySum = 0;
+        int buySum = 0;
+        String[] values = stringFromFile.split(COMA);
```

Comment:
instead of concatenating lines which you have read from a file to a string a then splitting them again by that coma, you should have just returned a string array and accepted it in this method or concatenated it with a `System.lineSeparator()` delimiter instead of coma in readFromFile. 
Because now you have a situation where you have a mess with your data, it was a clear lines separation which is handy to work with
```
supply,47
buy,33
result,14
```
but now you have this: 
`supply,47,buy,33,result,14`
and your for loop should not look like this, so change it to something like:
```
for (String line : lines) {    // lines is a String[]
    String[] data = line.split(COMA);
    data[0]; // operation type
    data[1]; // count
}
```
without any `i + 1`

Reply:
changed **readFromFile** output from concataneted String to String array and removed [i + 1] from loop.